File: src/main/scala/au/com/rainmore/centus/modules/core/BaseMongoRepository.scala
package au.com.rainmore.centus.modules.core

import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.querydsl.QuerydslPredicateExecutor
import org.springframework.data.repository.NoRepositoryBean

@NoRepositoryBean
trait BaseMongoRepository[T, ID]
    extends MongoRepository[T, ID]
    with QuerydslPredicateExecutor[T]

File: src/main/java/au/com/rainmore/centus/modules/core/BaseMongoRepository.java
package au.com.rainmore.centus.modules.core;

import au.com.rainmore.centus.modules.core.domains.Model;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;
import org.springframework.data.repository.NoRepositoryBean;

@NoRepositoryBean
public interface BaseMongoRepository<T extends Model, ID>
        extends MongoRepository<T, ID>, QuerydslPredicateExecutor<T> {
}
